pinecone/index_connection_test.go

```diff
@@ -991,6 +991,39 @@ func TestToUsageUnit(t *testing.T) {
 	}
 }
 
+func TestNormalizeHostUnit(t *testing.T) {
+	tests := []struct {
+		name         string
+		host         string
+		expectedHost string
+	}{
+		{
+			name:         "http:// scheme should be removed",
+			host:         "http://this-is-my-host.io",
+			expectedHost: "this-is-my-host.io:443",
+		}, {
+			name:         "https:// scheme should be removed",
+			host:         "https://this-is-my-host.io",
+			expectedHost: "this-is-my-host.io:443",
+		}, {
+			name:         "http:// scheme without a port should be removed",
+			host:         "http://this-is-my-host.io",
+			expectedHost: "this-is-my-host.io:443",
+		}, {
+			name:         "http:// scheme and port should be maintained",
+			host:         "http://this-is-my-host.io:8080",
+			expectedHost: "http://this-is-my-host.io:8080",
```

Comment:
It'd be nice to add an iteration here that does `https` with a port, too

Reply:
Yeah, I think I meant to add that case and kept one the same through copy pasta or something. 😅